lib/manager/argocd/extract.ts

```diff
@@ -0,0 +1,71 @@
+import * as gitTags from '../../datasource/git-tags';
+import * as helm from '../../datasource/helm';
+import type { ExtractConfig, PackageDependency, PackageFile } from '../types';
+import { fileTestRegex, keyValueExtractionRegex } from './util';
+
+function createDependency(
+  attributes: Record<string, string>
+): PackageDependency {
+  let result: PackageDependency;
+
+  if (attributes.repoURL == null || attributes.targetRevision == null) {
+    return null;
+  }
+
+  // a chart variable is defined this is helm declaration
+  if (attributes.chart) {
+    result = {
+      depName: attributes.chart,
+      registryUrls: [attributes.repoURL],
+      currentValue: attributes.targetRevision,
+      datasource: helm.id,
+    };
+  } else {
+    result = {
+      depName: attributes.repoURL,
+      currentValue: attributes.targetRevision,
+      datasource: gitTags.id,
+    };
+  }
+  return result;
+}
+
+export function extractPackageFile(
+  content: string,
+  fileName: string,
+  config?: ExtractConfig
+): PackageFile | null {
+  // check for argo reference. API version for the kind attribute is used
+  if (fileTestRegex.test(content) === false) {
```

Comment:
I would like to use yaml parser instead of regex here. There is also a function to parse multi yaml docs.